minicriu-client.c

```diff
@@ -285,6 +303,9 @@ static void mc_sighnd(int sig) {
 			: "memory");
 	}
 
+	__atomic_fetch_add(&mc_futex_checkpoint, 1, __ATOMIC_SEQ_CST);
```

Comment:
Let's introduce and use another variable. This is not checkpoint anymore, and renaming and using a counter for multiple purposes will make the code harder for understanding.